lms/djangoapps/support/static/support/jsx/entitlements/components/EntitlementForm/index.jsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Button, InputSelect, InputText, TextArea } from '@edx/paragon';
+
+
+class EntitlementForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const isReissue = props.entitlement !== null && props.entitlement !== undefined;
+
+    this.state = {
+      isReissue,
+      courseUuid: isReissue ? props.entitlement.courseUuid : '',
+      mode: isReissue ? props.entitlement.mode : '',
+      username: isReissue ? props.entitlement.user : '',
+      comments: 'Add any additional comments here',
+    };
+
+    this.onClose = this.onClose.bind(this);
+    this.handleCourseUUIDChange = this.handleCourseUUIDChange.bind(this);
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handleModeChange = this.handleModeChange.bind(this);
+    this.handleCommentsChange = this.handleCommentsChange.bind(this);
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  onClose() {
+    this.props.closeForm();
+  }
+
+  handleCourseUUIDChange(courseUuid) {
+    this.setState({ courseUuid });
+  }
+
+  handleUsernameChange(username) {
+    this.setState({ username });
+  }
+
+  handleModeChange(mode) {
+    this.setState({ mode });
+  }
+
+  handleCommentsChange(comments) {
+    this.setState({ comments });
+  }
+
+  submitForm() {
+    if (this.state.isReissue) { // if there is an active entitlement we are updating an entitlement
+      const { comments } = this.state;
```

Comment:
You destructure `this.state` in a couple places - maybe just do everything at the beginning of the function?

```javascript
const {
  isReissue,
  courseUuid,
  mode,
  comments,
} = this.state;

if (isReissue) {
  const { entitlement } = this.props;
  this.props.reissueEntitlement({ entitlement, comments });
} else {
  this.props.createEntitlement({ courseUuid, username, mode, comments });
}
```